### DISCUSSION TITLE
Connection is closed with ioredis module #6508

### DISCUSSION THREAD

**Discussion by sonlm** [Budibase/budibase#6536] [Help]
Hosting

- Self host on our server
- Method: GET /builder, env: k8s
- Budibase Version: 1.0.48
- Helm chart version: budibase-0.2.11
- Describe the bug
`ERROR [1656558037720] (30 on app-service-5f6ff4bb87-p79t6): Connection is closed. Error: Connection is closed. at Redis.sendCommand (/app/node_modules/ioredis/built/redis/index.js:628:24) at Redis.get (/app/node_modules/ioredis/built/commander.js:122:25) at RedisWrapper.<anonymous> (/app/node_modules/@budibase/backend-core/dist/src/redis/index.js:180:41) at Generator.next (<anonymous>) at /app/node_modules/@budibase/backend-core/dist/src/redis/index.js:8:71 at new Promise (<anonymous>) at __awaiter (/app/node_modules/@budibase/backend-core/dist/src/redis/index.js:4:12) at RedisWrapper.get (/app/node_modules/@budibase/backend-core/dist/src/redis/index.js:178:16) at /app/node_modules/@budibase/backend-core/dist/src/cache/generic.js:40:32 at Generator.next (<anonymous>) at fulfilled (/app/node_modules/@budibase/backend-core/dist/src/cache/generic.js:5:58)`
`Trace: Error: Connection is closed. at Redis.sendCommand (/app/node_modules/ioredis/built/redis/index.js:628:24) at Redis.get (/app/node_modules/ioredis/built/commander.js:122:25) at RedisWrapper.<anonymous> (/app/node_modules/@budibase/backend-core/dist/src/redis/index.js:180:41) at Generator.next (<anonymous>) at /app/node_modules/@budibase/backend-core/dist/src/redis/index.js:8:71 at new Promise (<anonymous>) at __awaiter (/app/node_modules/@budibase/backend-core/dist/src/redis/index.js:4:12) at RedisWrapper.get (/app/node_modules/@budibase/backend-core/dist/src/redis/index.js:178:16) at /app/node_modules/@budibase/backend-core/dist/src/cache/generic.js:40:32 at Generator.next (<anonymous>) at fulfilled (/app/node_modules/@budibase/backend-core/dist/src/cache/generic.js:5:58) at /app/dist/api/index.js:72:17 at Generator.throw (<anonymous>) at rejected (/app/dist/api/index.js:6:65)`

All pods are running

To Reproduce

Go to https://budibase.iviet.com/builder
See error: {"message":"Connection is closed.","status":500}
Expected behavior
Should show Login page

**Comment by melohagan** [comment]
Hi @sonlm 

That version of Budibase seems quite old. Is it possible for you to upgrade? 

The latest version is **1.0.214**

**Reply by sonlm** [reply]
Hi @melohagan 
I followed this docs: https://docs.budibase.com/docs/kubernetes-k8s , and this installed that version. So the docs was out up date ?

**Reply by sonlm** [reply]
https://github.com/Budibase/budibase/blob/v1.0.214/charts/budibase/Chart.yaml
Actually in the Chart.yaml, you will see appVersion: 1.0.48

**Reply by shogunpurple** [reply]
Hi @sonlm - that `appVersion` is actually a red herring. We inject the `latest` tag into our `values.yaml` here:
https://github.com/Budibase/budibase/blob/develop/charts/budibase/values.yaml#L90

You can confirm which version your images are using by running:
`kubectl describe deployment app-service -n budibase | grep Image`
then looking at the tag at the end of the image.

In this case, a restart of your redis and/or app-worker services might help. Could you try the following commands:
`kubectl rollout restart deployment app-service -n budibase`
`kubectl rollout restart deployment worker-service -n budibase`
`kubectl rollout restart deployment redis-service -n budibase`

Thanks!

**Reply by sonlm** [reply]
After making rollout restart commands, the ioredis is solved. Thank you. But another problem with couch db.
`Request URL: https://budibase.iviet.com/api/global/configs/checklist?tenantId=default
`
`{message: "Cannot connect to CouchDB instance", status: 500}
message: "Cannot connect to CouchDB instance"
status: 500`

I have tested from app-service pod
`root@app-service-569d9b54ff-mrmjv:/app# curl http://admin:***@budibase-svc-couchdb:5984
{"couchdb":"Welcome","version":"3.2.1","git_sha":"244d428af","uuid":"budibase-couchdb","features":["access-ready","partitioned","pluggable-storage-engines","reshard","scheduler"],"vendor":{"name":"The Apache Software Foundation"}}
`
`{"couchdb":"Welcome","version":"3.2.1","git_sha":"244d428af","uuid":"budibase-couchdb","features":["access-ready","partitioned","pluggable-storage-engines","reshard","scheduler"],"vendor":{"name":"The Apache Software Foundation"}}`
Seem like it's not a problem with connection.

**Reply by sonlm** [reply]
![Screenshot from 2022-07-04 17-46-10](https://user-images.githubusercontent.com/43743973/177139531-5bbe0777-f45a-4b07-8c27-0ce07fe52f6e.png)

**Reply by shogunpurple** [reply]
Hi @sonlm 

Hm, that's odd. Are you using the completely standard `values.yaml` provided with our helm charts? Or have you made any updates?

**Reply by sonlm** [reply]
I have edited some value, my custome values.yaml
`couchdb.dns.clusterDomainSuffix: local.jenkins-x-viettel.iviet.com` to be suitable with our cluster. So the couchdb is working, but budibase cannot connect to it.
As I see, in the environment variables, budibase use couch db url: http://budibase-svc-couchdb:5984
I have tested this domain in the pod, and it works. But budibase app cannot connect to it.
